network/phonebook.go

```diff
@@ -82,7 +105,48 @@ func (e *phonebookEntries) ReplacePeerList(they []string) {
 }
 
 func (e *phonebookEntries) updateRetryAfter(addr string, retryAfter time.Time) {
-	(*e)[addr] = phonebookData{retryAfter: retryAfter}
+	_, found := (*e)[addr]
+	if !found {
+		(*e)[addr] = phonebookData{retryAfter: retryAfter, recentConnectionTimes: make([]time.Time, 0)}
+	} else {
+		phbData := (*e)[addr]
+		phbData.retryAfter = retryAfter
+		(*e)[addr] = phbData
+	}
```

Comment:
nit: the code is correct, however, it could be more nicely written - 
```
  	entry := (*e)[addr]
  	entry.retryAfter = retryAfter
  	(*e)[addr] = entry
```
and it's functionally identical!